pump/storage/storage.go

```diff
@@ -1027,6 +1055,14 @@ func (a *Append) PullCommitBinlog(ctx context.Context, last int64) <-chan []byte
 			irange.Limit = encodeTSKey(atomic.LoadInt64(&a.maxCommitTS) + 1)
 			iter := a.metadata.NewIterator(irange, nil)
 
+			gcTS := atomic.LoadInt64(&a.gcTS)
+			if last < gcTS {
+				log.Warn("some binlogs was pugred, cause read from last ts is less than gcTS", zap.Int64("last ts", last), zap.Int64("gcTS", gcTS))
+				iter.Release()
+				a.responseBinlogError(ctx, responseChan, errors.Annotatef(butil.ErrorPurgedBinlog, "read binlog from last ts %d", last))
+				return
+			}
```

Comment:
Can we do this outside the `for` ?

Reply:
not for it, but can for the response error below